generated/typespec-ts/src/api/operations.ts

```diff
@@ -0,0 +1,1708 @@
+): Promise<CreateChatCompletionResponse> {
+  if (isUnexpected(result)) {
+    throw result.body;
+  }
+
+  return {
+    id: result.body["id"],
+    object: result.body["object"],
+    created: new Date(result.body["created"]),
+    model: result.body["model"],
+    choices: (result.body["choices"] ?? []).map((p) => ({
+      index: p["index"],
+      message: {
+        role: p.message["role"] as any,
+        content: p.message["content"],
+        functionCall: !p.message.function_call
+          ? undefined
+          : {
+              name: p.message.function_call?.["name"],
+              arguments: p.message.function_call?.["arguments"],
+            },
+      },
+      finishReason: p["finish_reason"] as any,
+    })),
+    usage: !result.body.usage
+      ? undefined
+      : {
+          promptTokens: result.body.usage?.["prompt_tokens"],
+          completionTokens: result.body.usage?.["completion_tokens"],
+          totalTokens: result.body.usage?.["total_tokens"],
+        },
+  };
+}
+
+export async function createChatCompletion(
+  context: Client,
+  body: CreateChatCompletionRequest,
+  options: CreateChatCompletionOptions = { requestOptions: {} }
+): Promise<CreateChatCompletionResponse> {
+  const result = await _createChatCompletionSend(context, body, options);
+  return _createChatCompletionDeserialize(result);
+}
+
+export function _createFineTuningJobSend(
+  context: Client,
+  job: CreateFineTuningJobRequest,
+  options: CreateFineTuningJobOptions = { requestOptions: {} }
+): StreamableMethod<
+  CreateFineTuningJob200Response | CreateFineTuningJobDefaultResponse
+> {
+  return context
+    .path("/fine_tuning/jobs")
+    .post({
+      ...operationOptionsToRequestParameters(options),
+      body: {
+        training_file: job["trainingFile"],
+        validation_file: job["validationFile"],
+        model: job["model"],
+        hyperparameters: !job.hyperparameters
+          ? undefined
+          : { n_epochs: job.hyperparameters?.["nEpochs"] },
+        suffix: job["suffix"],
+      },
+    });
+}
+
+export async function _createFineTuningJobDeserialize(
+  result: CreateFineTuningJob200Response | CreateFineTuningJobDefaultResponse
+): Promise<FineTuningJob> {
+  if (isUnexpected(result)) {
+    throw result.body;
+  }
+
+  return {
+    id: result.body["id"],
+    object: result.body["object"],
+    createdAt: new Date(result.body["created_at"]),
+    finishedAt: new Date(result.body["finished_at"]),
+    model: result.body["model"],
+    fineTunedModel: result.body["fine_tuned_model"],
+    organizationId: result.body["organization_id"],
+    status: result.body["status"] as any,
+    hyperparameters: {
+      nEpochs: result.body.hyperparameters["n_epochs"] as any,
+    },
+    trainingFile: result.body["training_file"],
+    validationFile: result.body["validation_file"],
+    resultFiles: result.body["result_files"],
+    trainedTokens: result.body["trained_tokens"],
+    error:
```

Comment:
[Compile Error]

src/api/operations.ts:357:5 - error TS2322: Type '{ message: any; code: any; param: any; } | null' is not assignable to type 'null'.
  Type '{ message: any; code: any; param: any; }' is not assignable to type 'null'.

357     error: